packages/@aws-cdk/aws-servicediscovery/lib/service.ts

```diff
@@ -0,0 +1,448 @@
+  }
+}
+
+function _getDnsRecords(dnsRecordType: DnsRecordType, dnsTtlSec?: number): DnsRecord[] {
+  const ttl = dnsTtlSec !== undefined ? dnsTtlSec.toString() : '60';
+
+  if (dnsRecordType === DnsRecordType.A_AAAA) {
+    return [{
+      type: DnsRecordType.A,
+      ttl
+    }, {
+      type: DnsRecordType.AAAA,
+      ttl,
+    }];
+  } else {
+    return [ { type: dnsRecordType, ttl} ];
+  }
+}
+
+/**
+ * An optional complex type that contains information about the DNS records that you want AWS Cloud Map to create when
+ * you register an instance.
+ */
+export interface DnsConfig {
+  /**
+   * List of DNS record that you want AWS Cloud Map to create when you register an instance.
+   */
+  dnsRecords: DnsRecord[];
+
+  /**
+   * The ID of the namespace that you want to use for DNS configuration.
+   */
+  namespaceId: string;
+
+  /**
+   * The routing policy that you want to apply to all Route 53 DNS records that AWS Cloud Map creates when you
+   * register an instance and specify this service.
+   */
+  routingPolicy?: RoutingPolicy;
+}
+
+/**
+ * Settings for one DNS record that you want AWS Cloud Map to create when you register an instance.
+ */
+export interface DnsRecord {
+  /**
+   * The record type
+   */
+  type: DnsRecordType;
+
+  /**
+   * The time to live for the record
+   */
+  ttl: string;
+}
+
+/**
+ * Settings for an optional Amazon Route 53 health check. If you specify settings for a health check, AWS Cloud Map
+ * associates the health check with all the records that you specify in DnsConfig. Only valid with a PublicDnsNamespace.
+ */
+export interface HealthCheckConfig {
+  /**
+   * The type of health check that you want to create, which indicates how Route 53 determines whether an endpoint is
+   * healthy. Cannot be modified once created. Supported values are HTTP, HTTPS, and TCP.
+   *
+   * @default HTTP
+   */
+  type?: HealthCheckType;
+
+  /**
+   * The path that you want Route 53 to request when performing health checks. Do not use when health check type is TCP.
+   *
+   * @default '/'
+   */
+  resourcePath?: string;
+
+  /**
+   * The number of consecutive health checks that an endpoint must pass or fail for Route 53 to change the current
+   * status of the endpoint from unhealthy to healthy or vice versa.
+   *
+   * @default 1
+   */
+  failureThreshold?: number;
+}
+
+/**
+ * Specifies information about an optional custom health check.
+ */
+export interface HealthCheckCustomConfig {
+  /**
+   * The number of 30-second intervals that you want Cloud Map to wait after receiving an
+   * UpdateInstanceCustomHealthStatus request before it changes the health status of a service instance.
+   *
+   * @default 1
+   */
+  failureThreshold?: number;
+}
+
+export enum DnsRecordType {
+  /**
+   * An A record
+   */
+  A = "A",
+
+  /**
+   * An AAAA record
+   */
+  AAAA = "AAAA",
+
+  /**
+   * Both an A and AAAA record
+   */
+  A_AAAA = "A, AAAA",
+
+  /**
+   * A Srv record
+   */
+  Srv = "SRV",
```

Comment:
Would call this `SRV` (because otherwise, why `AAAA`?)

Reply:
Oh. what is the convention here? I did think `Srv` and `Cname` looked weird but in some conversation I was told that was the Thing To Do :tm: